### PR TITLE
Fix(explosives): Selecting a cellphone as an IEDs trigger would throw an error

### LINKED ISSUE #5808: IED not linkable with cellphone
**Arma 3 Version:** `1.76.143187` (stable)
**CBA Version:** `3.4.1.170912` (stable)
**ACE3 Version:** `3.11.0` (stable e769d09)

**Mods:**
```
- CBA_A3
- ACE3
- ACEX
- RHSUSAF
- RHSAFRF
- RHSGREF
- RHSSAF
- BWMod
- Enhanced Movement
- task_force_radio
- WMO - Walkable Moving Objects
- Tanoa Tidesystem
- Stratis Tidesystem
- Altis Tidesystem
- ACE Compat - RHS: GREF
- ACE3 - BWMOD Compatibility
- ACE Compat - RHS Armed Forces of the Russian Federation
- ACE Compat - RHS United States Armed Forces
- Advanced Towing
- Advanced Sling Loading
- voyagerCompass HUD
```

**Description:**
- While linking an IED with a cellphone, the automatic adding of the speed dial code doesn't get added. 

**Steps to reproduce:**
- Place any IED
- Try to link it with a cellphone
- Dial with the cellphone

**Where did the issue occur?**
- Editor (Singleplayer)

**Placed Modules:**
- None

**RPT log file:**
- https://gist.github.com/jarckn/d831d5d6a111c50cedace1a24dd07a63

**Screenshots**
- [After linking IED with cellphone](https://abload.de/img/vlcsnap-2017-11-22-18k7zbe.png)
- [After starting dialing](https://abload.de/img/vlcsnap-2017-11-22-189hz6j.png)

### PR DESCRIPTION
**Pull request by bux**
**When merged this pull request will:**
- move `_code` var to outer scope, close #5808
- close cellphone dialog when being killed

### CODE REVIEW

**Comment by dedmen** [comment]
Btw I broke that private thingy 💃

File: addons/explosives/functions/fnc_onIncapacitated.sqf [hunk 1]
```diff
@@ -19,6 +19,9 @@
 params ["_unit"];
 TRACE_1("params",_unit);
 
+// close cellphone if open
+closeDialog 0;
```
Comment (PabstMirror) [inline addons/explosives/functions/fnc_onIncapacitated.sqf]: runs on all local units
`if (_unit == ace_player) then {`

Comment (bux) [reply addons/explosives/functions/fnc_onIncapacitated.sqf]: done

File: addons/explosives/functions/fnc_addCellphoneIED.sqf [hunk 2]
```diff
@@ -27,25 +27,30 @@ private _config = (_this select 3) select (count (_this select 3) - 1);
...
 {
     if !(_x in _detonators) exitWith{
         _hasRequired = false;
     };
 } count _requiredItems;
 
+private _code = "";
 private _codeSet = false;
+
 while {!_codeSet} do {
```
Comment (SilentSpike) [inline addons/explosives/functions/fnc_addCellphoneIED.sqf]: Alternative:
```sqf
private _code = "";
while {true} do {
    _code = [round(random 9999),4] call EFUNC(common,numberToDigitsString);
    if (([_code] call FUNC(getSpeedDialExplosive)) isEqualTo []) exitWith {};
};
```

**Pull request merged**